src/k8s-engine/lima.ts

```diff
@@ -1554,7 +1554,44 @@ export default class LimaBackend extends events.EventEmitter implements K8s.Kube
         ]);
         if (config.enabled) {
           this.lastCommandComment = 'Checking k3s images';
-          await this.progressTracker.action(this.lastCommandComment, 100, this.k3sHelper.ensureK3sImages(desiredVersion));
+          try {
+            await this.progressTracker.action(this.lastCommandComment, 100, this.k3sHelper.ensureK3sImages(desiredVersion));
+          } catch (ex:any) {
+            console.log(`Failed to find version ${ desiredVersion.raw }: ${ ex }`, ex);
+            if (ex.code === 'ECONNREFUSED' || ex.toString().includes('getaddrinfo ENOTFOUND github.com') || ex.toString().includes('getaddrinfo EAI_AGAIN github.com')) {
+              try {
+                const newVersion: semver.SemVer = await this.k3sHelper.selectClosestImage(desiredVersion);
+
+                if (semver.lt(newVersion, desiredVersion)) {
+                  const options: Electron.MessageBoxOptions = {
+                    message:   `Downgrading from ${ desiredVersion.raw } to ${ newVersion.raw } will lose Kubernetes workloads. OK to continue?`,
+                    type:      'question',
+                    buttons:   ['OK', 'Cancel'],
+                    defaultId: 1,
+                    title:     'Confirming migration',
+                    cancelId:  1
+                  };
+                  const mainWindow = getWindow('main');
+                  const result = await (mainWindow ? Electron.dialog.showMessageBox(mainWindow, options) : Electron.dialog.showMessageBox(options));
```

Comment:
We should be using a helper function to share code with the `show-message-box` IPC code.

Reply:
Yes, there's actually a benefit there.